x-pack/plugins/cloud_security_posture/public/pages/findings/latest_findings/latest_findings_container.tsx

```diff
@@ -47,25 +47,28 @@ export const LatestFindingsContainer = ({ dataView }: { dataView: DataView }) =>
     sort: urlQuery.sort,
   });
 
+  const findingsDistribution = getFindingsDistribution({
+    total: findingsGroupByNone.data?.total,
+    passed: findingsCount.data?.passed,
+    failed: findingsCount.data?.failed,
+    pageIndex: urlQuery.pageIndex,
+    pageSize: urlQuery.pageSize,
+    currentPageSize: findingsGroupByNone.data?.page.length,
+  });
+
   return (
     <div data-test-subj={TEST_SUBJECTS.FINDINGS_CONTAINER}>
       <FindingsSearchBar
         dataView={dataView}
         setQuery={setUrlQuery}
         query={urlQuery.query}
         filters={urlQuery.filters}
-        loading={findingsGroupByNone.isFetching}
+        loading={findingsCount.isFetching}
       />
       <PageWrapper>
         <LatestFindingsPageTitle />
         <FindingsGroupBySelector type="default" />
-        <FindingsDistributionBar
-          total={findingsGroupByNone.data?.total || 0}
-          passed={findingsCount.data?.passed || 0}
-          failed={findingsCount.data?.failed || 0}
-          pageStart={urlQuery.pageIndex * urlQuery.pageSize + 1} // API index is 0, but UI is 1
-          pageEnd={urlQuery.pageIndex * urlQuery.pageSize + urlQuery.pageSize}
-        />
+        {findingsDistribution && <FindingsDistributionBar {...findingsDistribution} />}
```

Comment:
we don't show this if there's no data to compute it. 

effectively, this only means renders without some results in cache. in practice, in means only initial renders would not show this and subsequent would (as we're using `keepPreviousData: true`)